pages/en/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -396,6 +394,99 @@ Off-chain state is not yet available.
+
+```ts
+class BurnExample extends SmartContract {
+...
+@method burnTokens(receiverAddress: PublicKey) {
+    this.token().burn({
+      address: receiverAddress,
+      amount: 1000000,
+    });
+  }
+}
+```
+
+In the snippet above, we define a smart contract called `BurnExample` with a method called `burnTokens`. Similar to minting, we use the `this.token()` property to call the `burn` method. This specifies the amount of tokens to burn for the specified address.
+
+**Note**: A zkApp cannot burn more tokens than the specified account has. If there is such a case, an error is thrown and no such transaction is made.
+
+##### Sending
+
+If two accounts hold a balance in a custom token, a zkApp must authorize such transaction. Sending tokens requires authorization from the zkApp, which is done by calling the `send` method on the `token` property. See the example below of how a zkApp can authorize sending tokens between two accounts:
+
+```ts
+class SendExample extends SmartContract {
+...
+@method sendTokens(
+    senderAddress: PublicKey,
+    receiverAddress: PublicKey,
+    amount: UInt64
+  ) {
+    this.token().send({
+      to: receiverAddress,
+      from: senderAddress,
+      amount,
+    });
+  }
+}
+```
+
+In the snippet above, we define a smart contract called `SendExample` with a method called `sendTokens`. In the same fashion as minting and burning, we use the `this.token()` property to call the `send` method.
+
+By defining a send method on our smart contract, the zkApp will fill in the authorization it needs to faciltate the transfer between two seperate accounts. Once that has been supplied, the two accounts simply add/subtract the specified amount from their respective balances.
+
+##### Understanding Important Terms
+
+If you are writing your own zkApp to interact with custom tokens, you should understand the following important terms:
+
+##### Token Accounts
+
+Token accounts are like regular accounts, but they hold a balance of a specific custom token instead of MINA. They are created from existing accounts and are specified by a public key and token id.
+If an existing account receives a transaction that is specified by a custom token, a token account for that public key and token id will be created if it does not exist.
+Token accounts are specific for each type of custom token, meaning that a single public key can have many different types of token accounts.
+
+To create a token account for a specified custom token, whenever an existing account receives a transaction denoted with a custom token, a token account is automatically created for that public key.
+
+Aside from sending custom tokens, custom tokens can be minted and burned by a **token owner account**. A token owner account is the governing zkApp account for a specific custom token.
```

Comment:
So there are two types of accounts? Shouldn't we call one token holder account and the other token owner account? (sorry if you already discussed this -- this is all brand new to me)

Reply:
Yup, that's right! I am trying to stick with the terminology we use in other parts of the protocol so I'm a little hesitant to introduce a new term like `token holder account`. I rephrased the content a bit to try and get the point across better.